server/service/endpoint_utils.go

```diff
@@ -23,7 +23,7 @@ import (
 	"github.com/gorilla/mux"
 )
 
-type handlerFunc func(ctx context.Context, request interface{}, svc fleet.Service) (interface{}, error)
+type handlerFunc func(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error)
```

Comment:
This is an awesome change and is an awesome step towards preventing some foot guns. I think we could do even better here by using the `Failer` interface defined in https://github.com/go-kit/kit/blob/60e8424101af501c525efaf67c0a2edf08667f80/endpoint/endpoint.go#L30 and removing `errorer` defined in https://github.com/fleetdm/fleet/blob/70b11518ae2641e0f6636972553013881bca8c2b/server/service/transport_error.go#L22. This wouldn't require that many more changes, but would prevent incorrectly returning an `error` type from an endpoint, which we also don't really catch.

Reply:
Good idea, but unless I'm missing something it requires some changes for our [encodeError](https://github.com/fleetdm/fleet/blob/main/server/service/transport.go#L36-L39) to continue behaving the same. (Plus IMO it's a feature we don't fully rely on go-kit to encode some stuff for us, the more control we have the better.)

Reply:
Am merging this as is and we can improve on top of this later.